src/adapters/utils/get-body-parser-options.util.ts

```diff
@@ -28,3 +30,63 @@ export function getBodyParserOptions<Options = NestHyperExpressBodyParserOptions
 
   return parserOptions;
 }
+
+function addToBufferMethod(stream: Readable): ReadableWithToBuffer {
+  const readableWithToBuffer = stream as ReadableWithToBuffer;
+
+  // Use an arrow function to ensure 'this' is the stream
+  readableWithToBuffer.toBuffer = async (): Promise<Buffer> => {
+    const chunks: Buffer[] = [];
+    for await (const chunk of readableWithToBuffer) {
+      chunks.push(Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk));
+    }
+    return Buffer.concat(chunks);
+  };
+
+  return readableWithToBuffer;
+}
+
+export async function multipartRequestBodyParser(
+  req: Request,
+): Promise<Record<string, MultipartFieldInterface>> {
+
+  const fields: Record<string, MultipartFieldInterface> = {};
+  await req.multipart(async (field) => {
+    if (field.file) {
+      const chunks: Buffer[] = [];
+      const fileStream = field.file.stream;
+
+      // Collect chunks from the file stream
+      for await (const chunk of fileStream) {
+        chunks.push(Buffer.from(chunk));
+      }
+      // Combine all chunks into a single buffer
+      const buffer = Buffer.concat(chunks);
+
+      fields[field.name] = {
```

Comment:
we can move common object before if/else

Reply:
done